apps/dashboard/src/utils/parseStepVariables.ts

```diff
@@ -127,32 +132,37 @@ export function parseStepVariables(schema: JSONSchemaDefinition, isEnhancedDiges
       if (typeof currentObj === 'boolean' || !('type' in currentObj)) return false;
 
       if (currentObj.type === 'array') {
-        const items = Array.isArray(currentObj.items) ? currentObj.items[0] : currentObj.items;
-        currentObj = items as JSONSchemaDefinition;
-        continue;
-      }
+        if (!currentObj.items) return false;
 
-      if (currentObj.type !== 'object') return false;
+        const items: JSONSchemaDefinition = Array.isArray(currentObj.items) ? currentObj.items[0] : currentObj.items;
+        if (typeof items === 'boolean') return false;
 
-      if (currentObj.additionalProperties === true) {
-        return true;
+        currentObj = items;
       }
 
-      if (!currentObj.properties || !(part in currentObj.properties)) {
+      if (typeof currentObj === 'boolean' || !('type' in currentObj)) return false;
+
+      if (currentObj.type === 'object') {
+        if (currentObj.additionalProperties === true) {
+          return true;
+        }
+
+        if (!currentObj.properties || !(part in currentObj.properties)) {
+          return false;
+        }
+
+        currentObj = currentObj.properties[part];
+      } else {
```

Comment:
There was a bug in how we check the props against a schema.

Btw we do this exact same thing on backend and its the #1 source of all bugs. I don't want to tackle it here in this PR but we need to figure out something else. Either using some library for schema parsing which allows some exceptions or something like that.

Reply:
@ChmaraX, you are spot on. In the other PR, I completely changed the logic of the `disallowed` function. PTAL [here](https://github.com/novuhq/novu/pull/8079/files#diff-ec09d146412ca39c7a2c821c1f9c30ce5e315c228afcb85810f72d5c1c6b0272R94).

The idea is, since we parse the variables from the API response and we know which ones are primitives(this needs to be renamed), arrays, namespaces, and such, the client validation logic must be much simpler, relying on the parsed results object. 

That is, there is no need to work with the Schema anymore.